File: src/my_project/medium_problems/from51to100/delete_operations_two_strings_dp.py
from typing import List 

class Solution:
    def minDistance(self, word1: str, word2: str) -> int:
        n, m = len(word1)+1, len(word2)+1
        dp = [[0] * m for _ in range(n)]
        for i in range(1, n):
            for j in range(1, m):
                if word1[i-1] == word2[j-1]:
                    dp[i][j] = 1 + dp[i-1][j-1]
                else:
                    dp[i][j] = max(dp[i-1][j], dp[i][j-1])
        return n + m - 2 - 2*dp[-1][-1]

File: src/my_project_typescript/medium_problems_typescript/from1to50/deleteOperationsTwoStringsDP.ts
function minDistanceTwo(word1: string, word2: string): number {

    let n: number = word1.length + 1 
    let m: number = word2.length + 1

    let dp: Array<Array<number>> =[]

    for (let j = 0; j < n; j++){
        dp.push(Array(m).fill(0))
    }

    for (let i = 1; i <n; i++){
        for (let j = 1; j < m; j++){
            if (word1[i-1] == word2[j-1]){
                dp[i][j] = 1 + dp[i-1][j-1]
            }
            else{
                dp[i][j] = Math.max(dp[i-1][j], dp[i][j-1])
            }
        }
    }

    return m + n -2 - (2*dp[n-1][m-1])

};

console.log(minDistanceTwo("leetcode","etco"))